src/features/Chat/components/ActiveWindow/index.tsx

```diff
@@ -0,0 +1,234 @@
+  const [sendMessage, { isLoading: isMessageSendLoading }] =
+    useSendMessageMutation();
+  const [updateChatStatus] = useUpdateStatusMutation();
+  const userId = useAppSelector(selectUserId);
+
+  const chat = useAppSelector(selectActiveChat);
+  const isLoadingMessages = useAppSelector(
+    selectIsLoadingBuddyMessages(chat?.buddyId),
+  );
+
+  const updateStatus = (status: ChatFolder) => {
+    if (!userId || !chat) return;
+    updateChatStatus({ userId, buddyId: chat.buddyId, status });
+  };
+
+  const handleMessageSend = (buddyId: string, text: string) => {
+    if (!userId || isMessageSendLoading) return;
+
+    const message = toSendMessage(buddyId, userId, text);
+    sendMessage({ userId, message });
+    setText('');
+  };
+
+  const isLoading = isLoadingMessages || isMessageSendLoading;
+
+  const getProfileIcon = (status: ChatFolder) => {
+    const icons = {
+      ok: <ProfileIcon color="purpleDark" />,
+      archived: <img src={ArchivedIcon} />,
+      banned: <img src={BlockedIcon} />,
+    };
+
+    return icons[status];
+  };
+
+  return (
+    chat && (
+      <ActiveChatContainer>
+        <HeaderBar>
+          <ProfileInfo>
+            {getProfileIcon(chat.status)}
+            <MentorName variant="h2">{chat.displayName}</MentorName>
+            <MentorBio variant="p">{chat.status}</MentorBio>
+          </ProfileInfo>
+          {showSearch ? (
+            <SearchBar>
+              <SearchInput
+                variant="iconInput"
+                color={searchValue ? 'blueDark' : 'greyFaded'}
+                leftIcon={{
+                  sizeInPx: searchInputIconSize,
+                  variant: 'search',
+                }}
+                rightButton={{
+                  onClick: () => setShowSearch(false),
+                  sizeInPx: closeInputIconSize,
+                  variant: 'closeWithBackground',
+                }}
+                onChange={setSearchValue}
+                placeholder={t('header.search')}
+                value={searchValue}
+              />
+            </SearchBar>
+          ) : (
+            <Buttons>
+              <IconButton
+                variant="search"
+                sizeInPx={24}
+                onClick={() => setShowSearch(true)}
+              />
+              {chat.status === 'ok' ? (
+                <>
+                  <StatusButton
+                    onClick={() => updateStatus('archived')}
+                    icon="archive"
+                    text={t('header.archive')}
+                  />
+                  <StatusButton
+                    onClick={() => updateStatus('banned')}
+                    icon="block"
+                    text={t('header.block')}
+                  />
+                </>
+              ) : (
+                <StatusButton
+                  onClick={() => updateStatus('ok')}
+                  icon="return"
+                  text={t('header.restore')}
+                />
+              )}
```

Comment:
Dont know, but maybe it could work better to render a list based on the status instead of ternary stuff. Not sure :p

Reply:
Or just abstract this as its own component, so there would not be too many conditionals in one return.. Again can be done in the next PR too